### PR TITLE
Added all static global, database, and table privileges

### PR DESCRIPTION
**Pull request by Hydrocharged**
This adds all static global, database, and table level privileges. They parse and persist but _do not_ yet restrict on anything. Future PR will dive into the full set of privilege checking.

One major point to note (which will probably bring about confusion otherwise) is that these new tables are implemented as essentially a "view" on top of the `mysql.user` table. This will _significantly_ help when it comes privilege checking, especially in the context of determining active roles and such. To complement this, `PrivilegeSet` was almost entirely rewritten. The related files are `data_editor_view.go` and `grant_table_shim.go`.

With the original direction, I could have 10 different table queries and 10 sets of privilege data to manage with only 2 roles. This way, I'll only have 3 queries and 3 sets of privilege data in the same scenario.

### CODE REVIEW

File: sql/grant_tables/grant_table_shim.go [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+package grant_tables
+
+import (
+	"io"
+
+	"github.com/dolthub/go-mysql-server/sql"
+	"github.com/dolthub/go-mysql-server/sql/in_mem_table"
+)
+
+type grantTableShim struct {
```
Comment (zachmu) [inline sql/grant_tables/grant_table_shim.go:24]: Should explain the purpose of this struct

File: sql/grant_tables/privilege_set.go [hunk 2]
```diff
@@ -0,0 +1,614 @@
...
+	i := 0
+	for _, dbSet := range ps.databases {
+		dbSets[i] = dbSet
+		i++
+	}
+	return dbSets
+}
+
+// Merge merges the given set of privileges to the calling set of privileges.
+func (ps PrivilegeSet) Merge(other PrivilegeSet) {
```
Comment (zachmu) [inline sql/grant_tables/privilege_set.go:201]: Probably should call this UnionWith

File: sql/grant_tables/privilege_set.go [hunk 3]
```diff
@@ -0,0 +1,614 @@
...
+	}
+	for _, otherTblSet := range otherPs.tables {
+		ps.getUseableTbl(otherTblSet.name).merge(otherTblSet)
+	}
+}
+
+// clear removes all database privileges.
+func (ps PrivilegeSetDatabase) clear() {
+	for priv := range ps.privs {
+		delete(ps.privs, priv)
```
Comment (zachmu) [inline sql/grant_tables/privilege_set.go:422]: Might just make a new map at this piont

File: sql/grant_tables/privilege_set.go [hunk 4]
```diff
@@ -0,0 +1,614 @@
...
+	i := 0
+	for _, colSet := range ps.columns {
+		colSets[i] = colSet
+		i++
+	}
+	return colSets
+}
+
+// Equals returns whether the given set of privileges is equivalent to the calling set.
+func (ps PrivilegeSetTable) Equals(otherPs PrivilegeSetTable) bool {
```
Comment (zachmu) [inline sql/grant_tables/privilege_set.go:474]: The presence of this makes me suspicious. Why is this needed?

File: sql/grant_tables/privilege_set.go [hunk 5]
```diff
@@ -0,0 +1,614 @@
...
+	i := 0
+	for priv := range ps.privs {
+		privs[i] = priv
+		i++
+	}
+	return privs
+}
+
+// ToSortedSlice returns all of the table privileges contained as a slice, sorted by their internal ID.
+func (ps PrivilegeSetTable) ToSortedSlice() []PrivilegeType {
```
Comment (zachmu) [inline sql/grant_tables/privilege_set.go:504]: This should be the default behavior, and if it's too slow we should change the data structures to make it faster

Never ever rely on hash ordering

**Review by zachmu — APPROVED** [review]
No major complaints, coming along pretty nicely

File: sql/grant_tables/grant_table_shim.go (hunk 1 above)
Comment (Hydrocharged) [reply sql/grant_tables/grant_table_shim.go:24]: Whoops forgot to add the comment

File: sql/grant_tables/privilege_set.go (hunk 3 above)
Comment (Hydrocharged) [reply sql/grant_tables/privilege_set.go:422]: This isn't a pointer method but a struct one. Without going into the details of exactly how it's beneficial for later, all of the `PrivilegeSet___` types are returned as structs so that their zero values behave the same as their fully-initialized counterparts. I played around with making valid nil pointer types, but wasn't a fan of the implementation. As a consequence, I can't replace the map, only modify it from here.

File: sql/grant_tables/privilege_set.go (hunk 4 above)
Comment (Hydrocharged) [reply sql/grant_tables/privilege_set.go:474]: This will be passed on to integrators, who may need convenience functions like that.

Comment (Hydrocharged) [reply sql/grant_tables/privilege_set.go:474]: Sometimes I do forget to remove a function that was needed in some pre-PR development, but at least for many of these "clearer" functions, they're often for integrators or they're mirroring other used functions (`Left()` is used and `Right()` isn't, but it makes more sense to have both).

File: sql/grant_tables/privilege_set.go (hunk 5 above)
Comment (Hydrocharged) [reply sql/grant_tables/privilege_set.go:504]: I'm not relying on hash ordering, and there are plenty of cases where it doesn't matter. For example, if I'm converting this set of privileges into a `sql.Row`, each privilege has its own defined column, so it doesn't matter if it's sorted or not. Sure sorting doesn't waste too much time, but in cases where it truly doesn't matter, I don't see the harm in having a non-sorted option.

Comment (Hydrocharged) [reply sql/grant_tables/privilege_set.go:504]: I've added to the function comment stating the above.
